src/hooks/useAtHgExecutionLimit.ts

```diff
@@ -0,0 +1,19 @@
+import { SubscriptionCodeType } from 'data/useGcom.types';
+
+import { useCurrentHGSubscription } from './useCurrentHGSubscription';
+import { useMonthlyTotalExecutionCount } from './useMonthlyTotalExecutionCount';
+
+export const FREE_EXECUTION_LIMIT = 100000;
+
+export function useAtHgExecutionLimit() {
+  const { data: totalMonthlyChecks, isLoading: isLoadingMonthlyChecks } = useMonthlyTotalExecutionCount();
+  const { data, isLoading: isLoadingHGSubscription } = useCurrentHGSubscription();
+  const isFree = data === SubscriptionCodeType.FREE;
```

Comment:
In the `grafanaBootstrapData` object, we have the `grafanaBootData.settings.buildInfo.edition` which for free users is `Cloud Free`. Have you explored reading this value from there instead calling the gcom api? It might not be enough, but worth to ask. 
<img width="438" alt="image" src="https://github.com/user-attachments/assets/775c9cad-1699-41b1-9a84-968cf7732367">

Reply:
There's also the `licenseInfo` object:
<img width="959" alt="image" src="https://github.com/user-attachments/assets/4503d373-f017-448a-8f26-e1c5fdaeb35b">

Reply:
Good spot! I've gone with this approach in the end. I thought long and hard about it and decided this was the right way to go after researching how the values get populated in core grafana. We still might want access to the gcom api directly at a later date and if that is the case we can bring back in the relevant code. I didn't see the need to leave it and complicate things as our memories get hazy and we forget why it is there 😅